<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mRender</class>
 <widget class="QMainWindow" name="mRender">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1121</width>
    <height>831</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>mRender</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="minimumSize">
    <size>
     <width>700</width>
     <height>500</height>
    </size>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QOpenGLWidget" name="placeholderWidget">
          <property name="minimumSize">
           <size>
            <width>300</width>
            <height>300</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>80</height>
           </size>
          </property>
         </spacer>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QPushButton" name="renderButton">
          <property name="text">
           <string>Render</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QTabWidget" name="tabWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>230</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_1">
           <attribute name="title">
            <string>Shape</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QRadioButton" name="squareRadioButton">
              <property name="text">
               <string>Square</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rectangleRadioButton">
              <property name="text">
               <string>Rectangle</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="pentagonRadioButton">
              <property name="text">
               <string>Pentagon</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="triangleRadioButton">
              <property name="text">
               <string>Triangle</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="trapRadioButton">
              <property name="text">
               <string>Trapezoid</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="hexRadioButton">
              <property name="text">
               <string>Hexagon</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="septRadioButton">
              <property name="text">
               <string>Septagon</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="octaRadioButton">
              <property name="text">
               <string>Octagon</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>Color</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QRadioButton" name="redRadioButton">
              <property name="text">
               <string>Red</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QRadioButton" name="orangeRadioButton">
              <property name="text">
               <string>Orange</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QRadioButton" name="blueRadioButton">
              <property name="text">
               <string>Blue</string>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="QRadioButton" name="magentaRadioButton">
              <property name="text">
               <string>Magenta</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QRadioButton" name="cyanRadioButton">
              <property name="text">
               <string>Cyan</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QRadioButton" name="yellowRadioButton">
              <property name="text">
               <string>Yellow</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QRadioButton" name="greenRadioButton">
              <property name="text">
               <string>Green</string>
              </property>
             </widget>
            </item>
            <item row="7" column="0">
             <widget class="QRadioButton" name="whiteRadioButton">
              <property name="text">
               <string>White</string>
              </property>
             </widget>
            </item>
            <item row="8" column="3">
             <widget class="QSlider" name="blueHorizontalSlider">
              <property name="styleSheet">
               <string notr="true">QSlider::handle::QSlider::horizontal{background-color: blue}</string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>256</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="8" column="1">
             <widget class="QSlider" name="redHorizontalSlider">
              <property name="styleSheet">
               <string notr="true">QSlider::handle::QSlider::horizontal{background-color: red}</string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>256</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="8" column="0">
             <widget class="QRadioButton" name="customColorRadioButton">
              <property name="text">
               <string>Custom</string>
              </property>
             </widget>
            </item>
            <item row="8" column="2">
             <widget class="QSlider" name="greenHorizontalSlider">
              <property name="styleSheet">
               <string notr="true">QSlider::handle::QSlider::horizontal{background-color: green}</string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>256</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="9" column="1" alignment="Qt::AlignHCenter|Qt::AlignTop">
             <widget class="QLabel" name="RLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>50</width>
                <height>19</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>130</width>
                <height>19</height>
               </size>
              </property>
              <property name="text">
               <string>1</string>
              </property>
             </widget>
            </item>
            <item row="9" column="2" alignment="Qt::AlignHCenter">
             <widget class="QLabel" name="GLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>50</width>
                <height>19</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>130</width>
                <height>19</height>
               </size>
              </property>
              <property name="text">
               <string>1</string>
              </property>
             </widget>
            </item>
            <item row="9" column="3" alignment="Qt::AlignHCenter">
             <widget class="QLabel" name="BLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>50</width>
                <height>19</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>130</width>
                <height>19</height>
               </size>
              </property>
              <property name="text">
               <string>1</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1121</width>
     <height>31</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>renderButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>render()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>340</x>
     <y>785</y>
    </hint>
    <hint type="destinationlabel">
     <x>111</x>
     <y>356</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>663</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>433</x>
     <y>321</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yellowRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>greenRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>373</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>360</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blueRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>513</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>whiteRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>653</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>magentaRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>583</y>
    </hint>
    <hint type="destinationlabel">
     <x>625</x>
     <y>915</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cyanRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>672</x>
     <y>443</y>
    </hint>
    <hint type="destinationlabel">
     <x>796</x>
     <y>874</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>orangeRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>612</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>568</x>
     <y>891</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>squareRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>791</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>285</x>
     <y>360</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rectangleRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>791</x>
     <y>235</y>
    </hint>
    <hint type="destinationlabel">
     <x>289</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>triangleRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>791</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>285</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pentagonRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>791</x>
     <y>319</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trapRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1088</x>
     <y>483</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hexRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>599</x>
     <y>561</y>
    </hint>
    <hint type="destinationlabel">
     <x>478</x>
     <y>754</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>octaRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1088</x>
     <y>717</y>
    </hint>
    <hint type="destinationlabel">
     <x>613</x>
     <y>879</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>septRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>mRender</receiver>
   <slot>updateShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1088</x>
     <y>639</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>780</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redHorizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>725</x>
     <y>704</y>
    </hint>
    <hint type="destinationlabel">
     <x>734</x>
     <y>853</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>greenHorizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>874</x>
     <y>706</y>
    </hint>
    <hint type="destinationlabel">
     <x>863</x>
     <y>852</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blueHorizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>mRender</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1006</x>
     <y>712</y>
    </hint>
    <hint type="destinationlabel">
     <x>1004</x>
     <y>860</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>customColorRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>mRender</receiver>
   <slot>toggleCustomColors(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>642</x>
     <y>722</y>
    </hint>
    <hint type="destinationlabel">
     <x>640</x>
     <y>871</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>render()</slot>
  <slot>updateShape()</slot>
  <slot>updateColor()</slot>
  <slot>toggleCustomColors(bool)</slot>
 </slots>
</ui>
